index.js

```diff
@@ -386,6 +410,122 @@ exports.getLastThreadPost = function (thread) {
   return msg
 }
 
+
+/* post revision utils */
+
+exports.getRevisions = function(ssb, thread, callback) {
+  function collectRevisions(thread, callback) {
+    // this function walks the revisions of a given thread, collecting them up
+    // asyncly
+    var deepThread = pluckRecursively(thread, 'related')[0]
+
+    var threadRevisionLogCB = multicb({pluck: 1})
+    deepThread
+    .filter(function(relatedMsg) { return isaRevisionTo(relatedMsg, thread) })
+    .forEach(function(edit) {
+      exports.createRevisionLog(ssb, edit, threadRevisionLogCB())
+    })
+    
+    threadRevisionLogCB(function(err, revLogs) {
+      if (err) callback(err)
+      const connectedLogs = revLogs
+        .filter(function(revLog) { 
+          // filter down to the log that links to this thread, by finding the log
+          // whose messages point to this thread as root 
+          // we should only need to test one message if createRevisionLog is
+          // passing tests
+          return revLog[revLog.length-1].key === thread.key
+        })
+
+      // the longest such log is the one that contains all of the revisions
+      const logLengths = connectedLogs.map(function(log) { return log.length })
+      const thisLog = connectedLogs.find(function(log) {
+                        return log.length === Math.max.apply(Math, logLengths)
+                      })
+      callback(null, thisLog)
```

Comment:
Can you explain why this search is occuring? You should be able to get the full revision log off of the thread root, by using `ssb.links()` or `ssb.relatedMessages()`, because all of the revisions have a `root` link to their top. You can then trust the order of those revisions by their sequence numbers, because they were all published by the same user.